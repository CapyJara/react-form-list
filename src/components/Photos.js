import React from 'react';
import Photo from './Photo';
import PropTypes from 'prop-types';

function Photos({ photos }) {
  return photos.map(photo => (
    <li key={photo.photo}>
      <Photo photo={photo.photo} />
    </li>
  )); 
}

Photos.propTypes = {
  photos: PropTypes.array.isRequired
};

export default Photos;


