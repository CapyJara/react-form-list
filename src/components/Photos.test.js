import React from 'react';
import { shallow } from 'enzyme';
import Photos from './Photos';

describe('Photos component', () => {
  it('renders a Photos', () => {
    const photos = [
      { photo: 'http://carrepairnearme.info/wp-content/uploads/2019/03/japanese-joinery-bed-woodworking-building-a-bed-for-my-boy-furniture-store-new-design-joinery-japanese-wood-joinery-bed.jpg' },
      { photo: 'https://i.pinimg.com/originals/29/cd/f5/29cdf57a4f9526cdd04eedf435712819.jpg' }
    ];

    const wrapper = shallow(<Photos photos={photos} />);
    expect(wrapper).toMatchSnapshot();
  });
});
