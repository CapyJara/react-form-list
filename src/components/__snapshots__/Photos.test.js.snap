// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Photos component renders a Photos 1`] = `
Array [
  <li
    key="http://carrepairnearme.info/wp-content/uploads/2019/03/japanese-joinery-bed-woodworking-building-a-bed-for-my-boy-furniture-store-new-design-joinery-japanese-wood-joinery-bed.jpg"
  >
    <Photo
      photo="http://carrepairnearme.info/wp-content/uploads/2019/03/japanese-joinery-bed-woodworking-building-a-bed-for-my-boy-furniture-store-new-design-joinery-japanese-wood-joinery-bed.jpg"
    />
  </li>,
  <li
    key="https://i.pinimg.com/originals/29/cd/f5/29cdf57a4f9526cdd04eedf435712819.jpg"
  >
    <Photo
      photo="https://i.pinimg.com/originals/29/cd/f5/29cdf57a4f9526cdd04eedf435712819.jpg"
    />
  </li>,
]
`;
